pkg/server/commands/batch_check_command.go

```diff
@@ -0,0 +1,162 @@
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+type BatchCheckQuery struct {
+	cacheController     cachecontroller.CacheController
+	checkResolver       graph.CheckResolver
+	datastore           storage.RelationshipTupleReader
+	logger              logger.Logger
+	maxChecksAllowed    uint32
+	maxConcurrentChecks uint32
+	typesys             *typesystem.TypeSystem
+}
+
+type BatchCheckCommandParams struct {
+	AuthorizationModelID string
+	Checks               []*openfgav1.BatchCheckItem
+	Consistency          openfgav1.ConsistencyPreference
+	StoreID              string
+}
+
+type BatchCheckOutcome struct {
+	CheckResponse *graph.ResolveCheckResponse
+	Duration      time.Duration
+	Err           error
+}
+
+type BatchCheckQueryOption func(*BatchCheckQuery)
+
+func WithBatchCheckCommandCacheController(cc cachecontroller.CacheController) BatchCheckQueryOption {
+	return func(bq *BatchCheckQuery) {
+		bq.cacheController = cc
+	}
+}
+
+func WithBatchCheckCommandLogger(l logger.Logger) BatchCheckQueryOption {
+	return func(bq *BatchCheckQuery) {
+		bq.logger = l
+	}
+}
+
+func WithBatchCheckMaxConcurrentChecks(maxConcurrentChecks uint32) BatchCheckQueryOption {
+	return func(bq *BatchCheckQuery) {
+		bq.maxConcurrentChecks = maxConcurrentChecks
+	}
+}
+
+func WithBatchCheckMaxChecksPerBatch(maxChecks uint32) BatchCheckQueryOption {
+	return func(bq *BatchCheckQuery) {
+		bq.maxChecksAllowed = maxChecks
+	}
+}
+
+func NewBatchCheckCommand(datastore storage.RelationshipTupleReader, checkResolver graph.CheckResolver, typesys *typesystem.TypeSystem, opts ...BatchCheckQueryOption) *BatchCheckQuery {
+	cmd := &BatchCheckQuery{
+		logger:              logger.NewNoopLogger(),
+		datastore:           datastore,
+		cacheController:     cachecontroller.NewNoopCacheController(),
+		checkResolver:       checkResolver,
+		typesys:             typesys,
+		maxChecksAllowed:    config.DefaultMaxChecksPerBatchCheck,
+		maxConcurrentChecks: config.DefaultMaxConcurrentChecksPerBatchCheck,
+	}
+
+	for _, opt := range opts {
+		opt(cmd)
+	}
+	return cmd
+}
+
+func (bq *BatchCheckQuery) Execute(ctx context.Context, params *BatchCheckCommandParams) (map[string]*BatchCheckOutcome, error) {
+	if len(params.Checks) > int(bq.maxChecksAllowed) {
+		return nil, errors.ValidationError(
+			fmt.Errorf("batchCheck received %d checks, the maximum allowed is %d ", len(params.Checks), bq.maxChecksAllowed),
+		)
+	}
+
+	if err := validateNoDuplicateCorrelationIDs(params.Checks); err != nil {
+		return nil, err
+	}
+
+	// This check query will be run against every check in the batch
+	checkQuery := NewCheckCommand(
+		bq.datastore,
+		bq.checkResolver,
+		bq.typesys,
+		WithCheckCommandLogger(bq.logger),
+		WithCacheController(bq.cacheController),
+	)
+
+	// the keys to this map are the correlation_id associated with each check
+	var resultMap = map[string]*BatchCheckOutcome{}
```

Comment:
Suggestion: `type correlationID string` and then `map[correlationId]` and you can get rid of the comment